src/main/java/seedu/module/model/ModuleBook.java

```diff
@@ -78,7 +79,14 @@ public boolean hasTask(Task task) {
      * The task must not already exist in the module book.
      */
     public void addTask(Task p) {
-        tasks.add(p);
+        assert(p != null);
+        if (ModuleManager.getListOfExistingModules().contains(p.getModule().toString())) {
+            assert(ModuleManager.moduleIsValid(p.getModule().toString()));
+            ModuleManager.insertTaskToMapping(p.getModule(), p);
+            tasks.add(p);
+        } else {
+
```

Comment:
yeah. This will only happen when u manually edit the json file with module codes which are not supported. So the tasks with invalid module codes will just be ignored